src/cobald/daemon/runners/trio_runner.py

```diff
@@ -1,47 +1,93 @@
-import trio
+from typing import Optional, Callable, Awaitable, Coroutine
+import asyncio
 from functools import partial
 
+import trio
+
+from .base_runner import BaseRunner, OrphanedReturn
+
 
-from .base_runner import BaseRunner
-from .async_tools import raise_return, AsyncExecution
+async def raise_return(payload: Callable[[], Awaitable]) -> None:
+    """Wrapper to raise exception on unhandled return values"""
+    value = await payload()
+    if value is not None:
+        raise OrphanedReturn(payload, value)
```

Comment:
The other implementations are handling this within the class specification. So for better maintainability I propose to move it into `TrioRunner`.